erpnext/accounts/doctype/process_payment_reconciliation/process_payment_reconciliation.py

```
@@ -0,0 +1,500 @@
+	):
+		frappe.throw(
+			_(
+				"Payment Reconciliation through backgound Job has been disabled. Enable it through {}"
+			).format(get_link_to_form("Accounts Settings", "Accounts Settings"))
+		)
+
+		return
+
+	if not is_scheduler_inactive():
+		if frappe.db.get_value("Process Payment Reconciliation", docname, "status") == "Queued":
+			frappe.db.set_value("Process Payment Reconciliation", docname, "status", "Running")
+			job_name = f"start_processing_{docname}"
+			if not is_job_running(job_name):
+				job = frappe.enqueue(
+					method="erpnext.accounts.doctype.process_payment_reconciliation.process_payment_reconciliation.reconcile_based_on_filters",
+					queue="long",
+					is_async=True,
+					job_name=job_name,
+					enqueue_after_commit=True,
+					doc=docname,
+				)
+				frappe.msgprint(_("Background Job {0} triggered").format(job_name))
+		elif frappe.db.get_value("Process Payment Reconciliation", docname, "status") == "Running":
+			# Resume tasks for running doc
+			job_name = f"start_processing_{docname}"
+			if not is_job_running(job_name):
+				job = frappe.enqueue(
+					method="erpnext.accounts.doctype.process_payment_reconciliation.process_payment_reconciliation.reconcile_based_on_filters",
+					queue="long",
+					is_async=True,
+					job_name=job_name,
+					doc=docname,
+				)
+				frappe.msgprint(_("Background Job {0} triggered").format(job_name))
+	else:
+		frappe.msgprint(_("Scheduler is Inactive. Can't trigger job now."))
+
+
+@frappe.whitelist()
+def trigger_reconciliation_for_queued_docs():
+	"""
+	Will be called from Cron Job
+	Fetch queued docs and start reconciliation process for each one
+	"""
+	if not frappe.db.get_single_value(
+		"Accounts Settings", "enable_payment_reconciliation_in_background"
+	):
+		frappe.throw(
+			_(
+				"Payment Reconciliation through backgound Job has been disabled. Enable it through {}"
+			).format(get_link_to_form("Accounts Settings", "Accounts Settings"))
+		)
```

Jobs will result in an error if settings is disabled, should simply return

---

If it just returns, the  Cron Job will show up as Completed. User will have no way to know that the feature was disabled. The `Process Payment Reconciliation` doc, which will be in queued status, will also not have any indication that Job didn't run due to disabled feature.

I think it's better to throw an exception here.